### PR TITLE
Require approvals for releasing repoman changes

### PR DESCRIPTION
**Pull request by danieljurek**
This PR adds: 

* Approval check before publishing changes to `main` 
* CI build (merges to `main` in the `azure-dev` repo) will push to a `staging` branch in the samples repo. Now you can get the latest templates by using the `staging` branch. 
* Diff comparisons for PRs that base off of the `staging` branch. This enables a dev to see only their changes w/r/t the `staging` branch instead of the full collection of changes changes compared to `main`. 
* Repoman - Ability to define a `--baseBranch`

### CODE REVIEW

**Comment by danieljurek** [comment]
Fixes #438

File: eng/pipelines/repoman.yml [hunk 1]
```diff
@@ -9,118 +9,164 @@ pr:
 
 variables:
   - template: /eng/pipelines/templates/variables/globals.yml
+  - name: StagingBranchName
```
Comment (danieljurek) [inline eng/pipelines/repoman.yml:12]: Re-add trigger for the CD path

File: generators/repo/src/commands/generate.ts [hunk 2]
```diff
@@ -129,7 +130,7 @@ export class GenerateCommand implements RepomanCommand {
     }
 
     private updateRemote = async (repo: GitRepo, remote: GitRemote): Promise<RemotePushResult> => {
-        const defaultBranch = remote.branch || "main";
+        const defaultBranch = this.options.baseBranch || remote.branch || "main";
```
Comment (wbreza) [inline generators/repo/src/commands/generate.ts:133]: If i recall, repoman supports having multiple remotes configured.  An optional implementation would be to configure multiple remotes to the same repo with different branches `main` vs `staging`.  Then we can use the `-r` flag to set the remote that should be the target for the generation.

**Review by wbreza — COMMENTED** [review]
Overall looks good.  Please review other comments and let me know if you have any questions.

File: templates/todo/projects/python-mongo/repo.yaml [hunk 3]
```diff
@@ -8,8 +8,11 @@ repo:
   includeProjectAssets: true
 
   remotes:
-    - name: azure-samples
+    - name: azure-samples-main
       url: git@github.com:Azure-Samples/todo-python-mongo.git
+    - name: azure-samples-main-staging
```
Comment (ellismg) [inline templates/todo/projects/python-mongo/repo.yaml:13]:
```suggestion
    - name: azure-samples-staging
```

**Review by ellismg — APPROVED** [review]

File: generators/repo/src/commands/generate.ts [hunk 4]
```diff
@@ -213,15 +213,16 @@ export class GenerateCommand implements RepomanCommand {
         console.info(chalk.white(`Cloning repo for remote...`));
         await repo.clone(remote.name, remote.url);
 
-        const isEmptyRepo = await repo.isEmptyRepo(remote.name);
+        const defaultBranchExists = await repo.remoteBranchExists(remote.name, defaultBranch);
 
-        if (isEmptyRepo) {
-            console.warn(chalk.yellowBright(`Remote is empty!`));
+        if (!defaultBranchExists) {
+            console.warn(chalk.yellowBright(`Remote does not have branch ${chalk.cyan(defaultBranch)}`));
```
Comment (wbreza) [inline generators/repo/src/commands/generate.ts:219]: Did you confirm this change still works on an empty repo without any branches?  This use case would be for a branch new repo that doesn't have any commits pushed yet.

**Review by wbreza — APPROVED** [review]
Looks good! Please verify all still works correct for empty repos since you made a change around that area.

File: generators/repo/src/commands/generate.ts (hunk 4 above)
Comment (danieljurek) [reply generators/repo/src/commands/generate.ts:219]:
```
...
************************************************************
************ Remote: djurek-empty-template-repo ************
************************************************************
Remote: https://github.com/danieljurek/test-empty-template-repo.git
Host: github.com
Org: danieljurek
Repo: test-empty-template-repo

Cloning repo for remote...
Remote does not have branch main
Creating default branch main...
Branch 'djurek/test-pr' doesn't exist on remote
Creating new branch: djurek/test-pr
Staging files from generated output

Found the following 91 change(s)... 
- A  .azdo/pipelines/README.md      
- A  .azdo/pipelines/azure-dev.yml  
- A  .devcontainer/Dockerfile       
- A  .devcontainer/devcontainer.json
- A  .gitattributes
... <files> 
- A  tests/playwright.config.ts
- A  tests/todo.spec.ts
Committing 91 change(s) to remote with message: Synchronize repo from Repoman

Pushing changes to djurek/test-pr...

Changes are available @ https://github.com/danieljurek/test-empty-template-repo/tree/djurek/test-pr
Compare @ https://github.com/danieljurek/test-empty-template-repo/compare/main...djurek/test-pr
```

**Pull request merged**